File: src/main/java/no/nav/bidrag/beregn/bidragsevne/periode/BidragsevnePeriode.java
package no.nav.bidrag.beregn.bidragsevne.periode;

import java.util.List;
import no.nav.bidrag.beregn.bidragsevne.beregning.BidragsevneBeregning;
import no.nav.bidrag.beregn.bidragsevne.bo.BeregnBidragsevneGrunnlag;
import no.nav.bidrag.beregn.bidragsevne.bo.BeregnetBidragsevneResultat;
import no.nav.bidrag.beregn.felles.bo.Avvik;

public interface BidragsevnePeriode {
    BeregnetBidragsevneResultat beregnPerioder(BeregnBidragsevneGrunnlag beregnBidragsevneGrunnlag);

    List<Avvik> validerInput(BeregnBidragsevneGrunnlag beregnBidragsevneGrunnlag);

    static BidragsevnePeriode getInstance() {
        return new BidragsevnePeriodeImpl(BidragsevneBeregning.getInstance());
    }
}

File: src/main/kotlin/no/nav/bidrag/beregn/bidragsevne/periode/BidragsevnePeriode.kt
package no.nav.bidrag.beregn.bidragsevne.periode

import no.nav.bidrag.beregn.bidragsevne.beregning.BidragsevneBeregning
import no.nav.bidrag.beregn.bidragsevne.bo.BeregnBidragsevneGrunnlag
import no.nav.bidrag.beregn.bidragsevne.bo.BeregnetBidragsevneResultat
import no.nav.bidrag.beregn.felles.bo.Avvik

interface BidragsevnePeriode {
    fun beregnPerioder(grunnlag: BeregnBidragsevneGrunnlag): BeregnetBidragsevneResultat
    fun validerInput(grunnlag: BeregnBidragsevneGrunnlag): List<Avvik>

    companion object {
        fun getInstance(): BidragsevnePeriode = BidragsevnePeriodeImpl(BidragsevneBeregning.getInstance())
    }
}
